### PR TITLE
Upgrade phpunit, switch CI, PHP8 Fix

### PR DESCRIPTION
**Pull request by baohx2000**
Upgraded phpunit, switch to github actions, fix for php 8. Take it or leave it.

I'll do another PR with just the php 8 fix.

### CODE REVIEW

File: phpunit.xml [hunk 1]
```diff
@@ -1,43 +1,26 @@
...
-            highLowerBound="70"
-            showUncoveredFiles="true"
-        />
-        -->
-    </logging>
+<!--    <filter>-->
+<!--        <whitelist addUncoveredFilesFromWhitelist="true">-->
+<!--            <directory>./lib</directory>-->
+<!--        </whitelist>-->
+<!--    </filter>-->
```
Comment (kelunik) [inline phpunit.xml:16]: This seems like a leftover, no?

File: test/Benchmark/TwoDeps.php [hunk 2]
```diff
@@ -0,0 +1,12 @@
+<?php
+
+
+namespace Auryn\Test\Benchmark;
+
+
+class TwoDeps
```
Comment (kelunik) [inline test/Benchmark/TwoDeps.php:7]:
```suggestion
<?php

namespace Auryn\Test\Benchmark;

class TwoDeps
```

**Comment by kelunik** [comment]
I forgot this exists and did the changes manually. -.-

https://github.com/amphp/injector/commit/d657e9d7a017ea498817c13f4f5cd4853cc21dee

**Comment by faizanakram99** [comment]
> I forgot this exists and did the changes manually. -.-
> 
> [amphp/injector@d657e9d](https://github.com/amphp/injector/commit/d657e9d7a017ea498817c13f4f5cd4853cc21dee)

Is this a fork of this repository ?
If so, why doesn't github show it as a fork ?

Also, what about this repository. Is this now unmaintained ?

In that case it would be better to archive this repository and used the one in amphp

**Comment by baohx2000** [comment]
It is a fork, but removed linkage is probably mainly due to namespace changes and default PHP version requirements without having done a major version bump. Hopefully someone can get a note and link into the readme on this repo. IIRC, there is an issue or comment somewhere here about moving to amphp.

**Comment by kelunik** [comment]
The API changed quite a lot, yeah, we'll add links once there's a new release.

**Comment by faizanakram99** [comment]
Ah ok, thanks

**Comment by kelunik** [comment]
Feedback on the new API is highly appreciated!

**Comment by Danack** [comment]
Thanks for the inspiration.....I eventually got around to upgrading things, but for reasons didn't accept the PR.

Probably going to be doing a 2.0 release 'soon'.

**Pull request closed without merging**